### PR TITLE
Implement XSS protection and refactor duplicate code in search and websiteuploader pages

### PR DESCRIPTION
**Pull request by B-Naoufal**

### CODE REVIEW

**Review by peterjah — APPROVED** [review]

File: pkg/dnshelper/dnshelper.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	}
+
+	websiteStorerAddress := websiteValue[indexOfWebsiteAddress]
+
+	websiteDescription := ""
+	if len(websiteValue) > indexOfWebsiteDescription {
+		websiteDescription = websiteValue[indexOfWebsiteDescription]
+	}
+	// Prevent XSS by escaping special characters in websiteDescription
+	escapedDescription := template.HTMLEscapeString(websiteDescription)
```
Comment (Thykof) [inline pkg/dnshelper/dnshelper.go:33]: this peace of code is not optimal: if `len(websiteValue) > indexOfWebsiteDescription` is false, then we will call `HTMLEscapeString` for nothing

Comment (Thykof) [reply pkg/dnshelper/dnshelper.go:33]: I would suggest to call `HTMLEscapeString` inside the if statement

**Review by Thykof — APPROVED** [review]

File: pkg/dnshelper/dnshelper.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+const (
+	metaKey = "META"
+
+	// Indexes of data in website name key.
+	indexOfWebsiteAddress     = 0 // Index of website Address in the dnsValue array
+	indexOfWebsiteDescription = 2 // Index of website Description in the dnsValue array
+)
+
+// Helper function to fetch website values from DNS entry.
```
Comment (gregLibert) [inline pkg/dnshelper/dnshelper.go:19]: As you function signature is ambiguous, you should describe the returned values explaining that the first one is an address and the second on is the description content.

Furthermore, you should avoid using word such as get or fecth we you don't grab the infomration from a remote service. here you are simply converting it.

Something as simple as the following should be sufficent:

```suggestion
// AddressAndDescription fetch the website address and it's description from the DNS entry.
```

File: pkg/dnshelper/dnshelper.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+func GetWebsiteValues(dnsValue []byte) (string, string, error) {
+	websiteValue := convert.ByteToStringArray(dnsValue)
```
Comment (gregLibert) [inline pkg/dnshelper/dnshelper.go:21]: You use websiteValue a lot. You should clarify what your are manipulating using a clearer functin name
```suggestion
	dnsRecords := convert.ByteToStringArray(dnsValue)
```

File: pkg/dnshelper/dnshelper.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	if len(websiteValue) <= indexOfWebsiteAddress {
+		return "", "", fmt.Errorf("invalid website value: missing website address")
+	}
+
+	websiteStorerAddress := websiteValue[indexOfWebsiteAddress]
```
Comment (gregLibert) [inline pkg/dnshelper/dnshelper.go:26]: Same, we alreadu know that the this is a DNS for a website. you can simply the varaible name:
```suggestion
	address := websiteValue[indexOfWebsiteAddress]
```

File: pkg/dnshelper/dnshelper.go (hunk 1 above)
Comment (gregLibert) [inline pkg/dnshelper/dnshelper.go:33]: Same as @Thykof with a clearer variable name:
```suggestion
	descrition := ""
	if len(websiteValue) > indexOfWebsiteDescription {
	// Prevent XSS by escaping the HTML content.
	// see https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#output-encoding-for-html-contexts
		description = template.HTMLEscapeString(websiteDescription)
	}
```

File: pkg/dnshelper/dnshelper.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	if len(websiteValue) > indexOfWebsiteDescription {
+		websiteDescription = websiteValue[indexOfWebsiteDescription]
+	}
+	// Prevent XSS by escaping special characters in websiteDescription
+	escapedDescription := template.HTMLEscapeString(websiteDescription)
+
+	return websiteStorerAddress, escapedDescription, nil
+}
+
+// Helper function to retrieve website metadata.
```
Comment (gregLibert) [inline pkg/dnshelper/dnshelper.go:38]:
```suggestion
// GetWebsiteMetadata retrieves candidate metadata of the website.
```

File: pkg/dnshelper/dnshelper.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+const (
+	metaKey = "META"
+
+	// Indexes of data in website name key.
+	indexOfWebsiteAddress     = 0 // Index of website Address in the dnsValue array
+	indexOfWebsiteDescription = 2 // Index of website Description in the dnsValue array
+)
+
+// Helper function to fetch website values from DNS entry.
+func GetWebsiteValues(dnsValue []byte) (string, string, error) {
```
Comment (gregLibert) [inline pkg/dnshelper/dnshelper.go:20]: Can you please unit test this function to be sure that the XSS attack is indeed aborted?

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: pkg/dnshelper/dnshelper.go [hunk 7]
```diff
@@ -16,32 +16,38 @@ const (
...
-	websiteStorerAddress := websiteValue[indexOfWebsiteAddress]
+	address := dnsRecords[indexOfWebsiteAddress]
 
-	escapedDescription := ""
-	if len(websiteValue) > indexOfWebsiteDescription {
-		websiteDescription := websiteValue[indexOfWebsiteDescription]
+	description := ""
+
+	if len(dnsRecords) > indexOfWebsiteDescription {
+		websiteDescription := dnsRecords[indexOfWebsiteDescription]
```
Comment (gregLibert) [inline pkg/dnshelper/dnshelper.go:34]:
```suggestion
		unsafeDescription := dnsRecords[indexOfWebsiteDescription]
```

**Review by gregLibert — APPROVED** [review]
One facultative comment to clarify that the content of the DNS should not be trusted. It's optional.

**Pull request merged**